core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -5222,6 +5222,7 @@ object Stream extends StreamLowPriority {
       def handleErrorWith[A](s: Stream[F, A])(h: Throwable => Stream[F, A]) =
         s.handleErrorWith(h)
       def raiseError[A](t: Throwable) = Stream.raiseError[F](t)
+      override def map[A, B](fa: Stream[F, A])(f: A => B): Stream[F, B] = fa.map(f)
```

Comment:
It's unfortunate that this fix has to be duplicated. Perhaps we should extract a:
```scala
class StreamMonad[F] extends Monad[Stream[F, *]] {
  ...
}
```
Then the code can be shared.

Reply:
Now my understanding is:
After extracting a class like
```scala
class StreamMonad[F] extends Monad[Stream[F, *]] {
  ...
}
```
the original code
```scala
private[fs2] trait StreamLowPriority {
  implicit def monadInstance[F[_]]: Monad[Stream[F, *]] =
    new Monad[Stream[F, *]] {
      ...
}
```
will become like below?
```scala
private[fs2] trait StreamLowPriority {
  implicit def monadInstance[F[_]]: StreamMonad[Stream[F, *]] =
    new StreamMonad[Stream[F, *]] {
      ...
    }
}
```

Reply:
```scala
implicit def monadErrorInstance[F[_]]: MonadError[Stream[F, *], Throwable] =
new StreamMonad[F] with MonadError[Stream[F, *], Throwable] {
  // inherit monad methods, implement monad error methods
}

Reply:
Thanks! It's very helpful.😊

Reply:
@armanbilge IMHIO it is always good practice to give explicit names to long classes, just as it is good practice to name long lambdas. @yyy1000 Just remember to keep it private to the `Stream` companion object.